src/cregistry/entry.c

```diff
@@ -517,16 +517,45 @@ int reg_entry_installed(reg_registry* reg, char* name, reg_entry*** entries,
  *
  * @param [in] reg     registry to search in
  * @param [in] path    path of the file to check ownership of
+ * @param [in] cs      false if check should be performed case-insensitive,
+ *                     true otherwise
  * @param [out] entry  the owner, or NULL if no active port owns the file
  * @param [out] errPtr on error, a description of the error that occurred
  * @return             true if success; false if failure
  */
-int reg_entry_owner(reg_registry* reg, char* path, reg_entry** entry,
+int reg_entry_owner(reg_registry* reg, char* path, int cs, reg_entry** entry,
         reg_error* errPtr) {
     int result = 0;
     sqlite3_stmt* stmt = NULL;
-    char* query = "SELECT id FROM registry.files WHERE actual_path=? AND active";
     int lower_bound = 0;
+    const char* base_query_start = "SELECT id FROM registry.files WHERE actual_path";
+    const char* base_query_mid = "?";
+    const char* base_query_end = " AND active";
+    const char* cis_operator = " LIKE ";
+    const char* cs_operator = "=";
+    const char* extend_query = " COLLATE NOCASE";
```

Comment:
It works, but it doesn't help either. For some reason `COLLATE NOCASE` doesn't seem to change anything.

I've played around a bit with the sqlite3 binary on my 10.6 VM directly and it seems that "COLLATE NOCASE" must be carefully placed to have an effect.

Case in point:
```
sqlite> SELECT id from files WHERE actual_path="/opt/local/bin/gsed" AND active COLLATE NOCASE;
50
sqlite> SELECT id from files WHERE actual_path="/opt/local/bin/gseD" AND active COLLATE NOCASE;
sqlite> SELECT id from files WHERE actual_path="/opt/local/bin/gseD" COLLATE NOCASE AND active;
50
sqlite> SELECT id from files WHERE UPPER(actual_path)="/opt/local/bin/gseD" COLLATE NOCASE AND active;
50
sqlite> SELECT id from files WHERE (UPPER(actual_path)="/opt/local/bin/gseD" COLLATE NOCASE) AND active;
50
```

But if we're using `LIKE` anyway, I can drop the COLLATE NOCASE thingy.